episodes/02-data-structures.Rmd

```diff
@@ -0,0 +1,235 @@
+teaching: 10
+exercises: 2
+---
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- What are the basic data types in R?
+- How do I represent categorical information in R?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- To be aware of the different types of data.
+- To begin exploring data frames, and understand how they are related to vectors, factors and lists.
+- To be able to ask questions from R about the type, class, and structure of an object.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Vectors 
+So far we've looked on individual values. Now we will move to a data structure 
+called vectors. Vectors are arrays of values of the same data type. 
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: callout
+
+### Data types 
+
+Data type refers to a type of information that is stored by a value. 
+It can be:
+
+- `numerical` (a number)
+- `integer` (a number without information about decimal points)
+- `logical` (a boolean - are values TRUE or FALSE?)
+- `character` (a text/ string of characters)
+- `complex` (a complex number)
+- `raw` (raw bytes)
+
+We won't discuss `complex` or `raw` data type in the workshop.
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: callout
+
+### Data structures
+
+Vectors are the most common and basic data structure in R but you will come 
+across other data structures such as data frames, lists and matrices as well.
+In short:
+
+- data.frames is a two-dimensional data structure in which columns are vectors of the same length that can have different data types. We will use this data structure in this lesson.
+- lists can have an arbitrary structure and can mix data types;
+- matrices are two-dimensional data structures containing elements of the same data type.
+
+For a more detailed description, see [Data Types and Structures](https://swcarpentry.github.io/r-novice-inflammation/13-supp-data-structures.html).
+
+Note that vector data in the geospatial context is different from vector data types. More about vector data in a [later lesson](../episodes/09-open-and-plot-vector-layers.Rmd)!
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+You can create a vector with a `c()` function. 
+
+```{r vectors}
+
+numeric_vector <- c(2, 6, 3) # vector of numbers - numeric data type.
+numeric_vector
+
+character_vector <- c('banana', 'apple', 'orange') # vector of words - or strings of characters- character data type
+character_vector
+
+logical_vector <- c(TRUE, FALSE, TRUE) # vector of logical values (is something true or false?)- logical data type.
+logical_vector
+
+```
+
+### Combining vectors 
+
+The combine function, `c()`, will also append things to an existing vector:
+
+```{r combine-vectors}
+
+ab_vector <- c('a', 'b')
+ab_vector
+
+abcd_vector <- c(ab_vector, 'c', 'd')
+abcd_vector
+
+```
+
+### Missing values  
+
```

Comment:
```suggestion

### Exercise 
<div class="alert alert-info">
<strong>Challenge</strong> 
Combine the abcd_vector with the numerical_vector in R. What is the datatype of this new vector and why?


### Missing values  

```

Reply:
Thanks, Kyri. I transformed this into a challenge with a hidden solution.